test/unit/strategies/BaseStrategyUnit.t.sol

```diff
@@ -0,0 +1,260 @@
+    }
+
+    function test___BaseStrategy_initShouldCallOnlyAllo(uint256 _poolId) external {
+        vm.assume(_poolId != 0);
+        baseStrategy.mock_call__checkOnlyAllo();
+
+        // It should call onlyAllo
+        baseStrategy.expectCall__checkOnlyAllo();
+
+        baseStrategy.call___BaseStrategy_init(_poolId);
+    }
+
+    function test___BaseStrategy_initRevertWhen_PoolIdIsZero(uint256 _currentPoolId, uint256 _poolId) external {
+        vm.assume(_currentPoolId != 0);
+        baseStrategy.mock_call__checkOnlyAllo();
+
+        baseStrategy.set__poolId(_currentPoolId);
+
+        // It should revert
+        vm.expectRevert(IBaseStrategy.BaseStrategy_ALREADY_INITIALIZED.selector);
+
+        baseStrategy.call___BaseStrategy_init(_poolId);
+    }
+
+    function test___BaseStrategy_initRevertWhen_PoolIdArgumentIsZero() external {
+        baseStrategy.mock_call__checkOnlyAllo();
+
+        // It should revert
+        vm.expectRevert(IBaseStrategy.BaseStrategy_INVALID_POOL_ID.selector);
+
+        baseStrategy.call___BaseStrategy_init(0);
+    }
+
+    function test___BaseStrategy_initShouldSetPoolId(uint256 _poolId) external {
+        vm.assume(_poolId != 0);
+        baseStrategy.mock_call__checkOnlyAllo();
+
+        baseStrategy.call___BaseStrategy_init(_poolId);
+
+        // It should set poolId
+        assertEq(baseStrategy.getPoolId(), _poolId);
+    }
+
+    function test_IncreasePoolAmountShouldCallOnlyAllo(uint256 _amount) external {
+        baseStrategy.mock_call__checkOnlyAllo();
+
+        // It should call onlyAllo
+        baseStrategy.expectCall__checkOnlyAllo();
+
+        baseStrategy.increasePoolAmount(_amount);
+    }
+
+    function test_IncreasePoolAmountShouldCall_beforeIncreasePoolAmount(uint256 _amount) external {
+        baseStrategy.mock_call__checkOnlyAllo();
+
+        // It should call _beforeIncreasePoolAmount
+        baseStrategy.expectCall__beforeIncreasePoolAmount(_amount);
+
+        baseStrategy.increasePoolAmount(_amount);
+    }
+
+    function test_IncreasePoolAmountShouldAddAmountToPoolAmount(uint256 _previousAmount, uint256 _amount) external {
+        vm.assume(_amount < type(uint256).max - _previousAmount);
+        baseStrategy.mock_call__checkOnlyAllo();
+        baseStrategy.set__poolAmount(_previousAmount);
+
+        baseStrategy.increasePoolAmount(_amount);
+
+        // It should add amount to poolAmount
+        assertEq(baseStrategy.getPoolAmount(), _previousAmount + _amount);
+    }
+
+    function test_IncreasePoolAmountShouldCall_afterIncreasePoolAmount(uint256 _amount) external {
+        baseStrategy.mock_call__checkOnlyAllo();
+
+        // It should call _afterIncreasePoolAmount
+        baseStrategy.expectCall__afterIncreasePoolAmount(_amount);
+
+        baseStrategy.increasePoolAmount(_amount);
+    }
+
+    function test_WithdrawWhenParametersAreValid(address _token, uint256 _amount, address _recipient) external {
+        // TODO: Fix the bug and fix this test
```

Comment:
What is the issue here?

Reply:
I'm referring to [this](https://github.com/allo-protocol/allo-v2.1/pull/17) bug